### ISSUE TITLE
Suggestion: Pre-Toggle Formatting (w/o selection)

### ISSUE THREAD

**Issue by metal450** [giuspen/cherrytree#1108]
Typically, text editors allow you to toggle on a given format, then begin typing.  For instance, you place the cursor, hit Ctrl+B to enable Bold, then you begin typing, & the text will come out in bold. You can give it a try right here in Github as an example.

In CT, it's necessary to first have some text selected before it will allow you to set a format.  In addition to leading to a less streamlined workflow even in ideal cases, a deeply ingrained familiarity with other text editors frequently leads to:
1) I try to enable a format via hotkey
2) "Warning - No Text is Selected" dialog is shown.  Ah, now I remember.
3) Dismiss the dialog & type the text
4) Go back & select the text
5) Enable the format
6) Return the cursor to after the text, so I can continue typing

Suggestion: it would be great if CT could support just toggling on the format, and then beginning to type.  This would avoid the need to go back & change the cursor position twice for each format-change :)

**Comment by txe** [comment]
If cursor is on a word or next to it, then Ctrl+B will select the whole word itself and will change bold for it.
I only find inconvenient that selection is left after changing bold. That doesn't have any purpose and makes user to press array buttons to remove selection.

**Comment by metal450** [comment]
Yeah, if there's a word already there, it works fine.  I'm referring to the scenario where you're about to start typing.  i.e. put the cursor on a new line; now I want to start typing in bold.  So in most editors, I press Ctrl+B, it switches to bold, then I start typing.  But instead, I have to type a word first.

I see that you can start typing, then change to bold, then move the cursor back to the end & continue typing.  But it's a bit counter-intuitive as it's inconsistent with other text editors.  i.e. here on Github, put the cursor in a brand new comment with no text, & hit Ctrl+i.  That switches it to italic for any text you start typing from that point. Does that make sense?

**Comment by txe** [comment]
Well, github editor maybe is not a good example, but I see your point.

**Comment by metal450** [comment]
Sure, I just mentioned it because it's something I know u have easy access too ;)  U could also try i.e. OneNote, Word, Outlook, WPS Office, TextMaker, Evolution, Wordpress, etc :)
